src/WorkspaceService.ts

```diff
@@ -73,23 +73,18 @@ export class WorkspaceService implements vscode.DocumentFormattingEditProvider {
       return [];
     }
 
+    const configFiles = await vscode.workspace.findFiles(DPRINT_CONFIG_FILEPATH_GLOB);
+
+    // Initialize the workspace folders with each sub configuration that's found.
     for (const folder of vscode.workspace.workspaceFolders) {
-      this.#folders.push(
-        new FolderService({
-          workspaceFolder: folder,
-          configUri: undefined,
-          outputChannel: this.#outputChannel,
-        }),
-      );
-
-      // now search within sub directories to find any configuration files
-      const subConfigFiles = await vscode.workspace.findFiles(`*/**/${DPRINT_CONFIG_FILENAME_GLOB}`);
```

Comment:
By the way, this was incredibly inefficient what it was previously doing here. Thanks for this change!